thrift0.13/src/main/java/com/linecorp/armeria/internal/common/thrift/ThriftServiceMetadata.java

```diff
@@ -225,6 +232,21 @@ private void registerFunction(Class<?> iface, String name,
      */
     @Nullable
     public ThriftFunction function(String method) {
-        return functions.get(method);
+        ThriftFunction func = functions.get(method);
+        if (func != null) {
+            return func;
+        }
+
+        // check method and convert to underscored style
+        method = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, method);
+        func = functions.get(method);
+        if (func != null) {
+            functionsLock.lock();
+            // add it to function, for next call
+            functions.putIfAbsent(method, func);
```

Comment:
Can we move this logic to `registerFunction` to avoid additional lock?

Reply:
If move this to `registerFunction` there two choice for me
1. use `Iface` method as the only key in `functions`.  Risk is I don't know what impact will happen.
2. use `Iface` method and `processMap` both as the key for `functions`. No potential risk, I need find a way to do this.

Reply:
I think it's better putting both full camel and original method names to the map (option 2).